src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_validators.py

```diff
@@ -0,0 +1,96 @@
+            raise CLIError('The registry name must be between 5 and 60 characters.')
+
+        p = re.compile('^([A-Za-z0-9]+)$')
+
+        if not p.match(registry_name):
+            raise CLIError('The registry name can contain only letters and numbers.')
+
+def validate_registry_name_create(namespace):
+    if namespace.registry_name:
+        client = get_acr_service_client()
+
+        result = client.operation.check_name_availability(
+            RegistryNameCheckRequest(
+                namespace.registry_name,
+                RESOURCE_TYPE
+            )
+        )
+
+        if not result.name_available: #pylint: disable=E1101
+            raise CLIError(result.message) #pylint: disable=E1101
+
+def validate_storage_account_name(namespace):
+    client = storage_client_factory().storage_accounts
+
+    if namespace.storage_account_name:
+        storage_account_name = namespace.storage_account_name
+        if arm_get_storage_account_by_name(storage_account_name) is None:
+            logger.warning('Command to create a storage account:')
+            logger.warning(
+                '  az storage account create ' +\
+                '-n <name> -g <resource-group> -l <location> --sku Standard_LRS')
+            logger.warning(
+                'The container registry must be at the same location as the storage account.')
+            raise CLIError(
+                'The storage account {} does not exist in the current subscription.'\
+                .format(storage_account_name))
+    else:
+        while True:
+            storage_account_name = str(uuid.uuid4()).replace('-', '')[:24]
+            if client.check_name_availability(storage_account_name).name_available is True: #pylint: disable=E1101
+                namespace.storage_account_name = storage_account_name
+                logger.warning(
+                    'New storage account with name %s will be created and used.',
+                    storage_account_name)
+                return
+
+def validate_resource_group_name(namespace):
+    if namespace.resource_group_name:
+        client = get_arm_service_client()
+        resource_group_name = namespace.resource_group_name
+
+        if not client.resource_groups.check_existence(resource_group_name):
+            logger.warning('Command to create a resource group:')
+            logger.warning('  az resource group create -n <name> -l <location>')
+            raise CLIError(
+                'The resource group {} does not exist in the current subscription.'\
+                .format(resource_group_name))
+
+def validate_password(namespace):
+    if namespace.password and not namespace.new_sp:
+        raise CLIError('--password has to be used with --new-sp.')
+
+def validate_role(namespace):
+    if namespace.role and not namespace.role.lower() in ALLOWED_ROLES:
+        raise CLIError('The role {} is not allowed. Allowed roles (Owner, Contributor, Reader).'\
```

Comment:
Suggest instead of hard-coding the string '(Owner, Contributor, Reader)', could do `', '.join(ALLOWED_ROLES)`